### PR TITLE
Adding single run option; Making it closer functionally to shipper.

### PR DESCRIPTION
**Pull request by bwplotka**
Related: https://github.com/observatorium/thanos-replicate/issues/7

* Changed important log lines to Info
* Added check if ext lset are set for Thanos.
* Added single-run option.

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
Nooo idea `make lint` runs locally.. @brancz any ideas? (:

**Comment by bwplotka** [comment]
PTAL @brancz @metalmatze

**Comment by brancz** [comment]
Just a bit, generally lgtm

File: replicate.go [hunk 1]
```diff
@@ -21,16 +22,21 @@ import (
...
 
 	httpMetricsBindAddr := regHTTPAddrFlag(cmd)
 
+	// TODO(bwplotka): Add support for local filesystem bucket implementation.
 	fromObjStoreConfig := regCommonObjStoreFlags(cmd, "from", false)
 	toObjStoreConfig := regCommonObjStoreFlags(cmd, "to", false)
 
 	matcherStrs := cmd.Flag("matcher", "Only blocks whose labels match this matcher will be replicated.").PlaceHolder("key=\"value\"").Strings()
 
+	singleRun := cmd.Flag("single-run", "Run replication only one time, then exit.").Default("false").Bool()
```
[the remark this answers is not in the document]
Comment (squat) [reply replicate.go]: In systemd the term is “one-shot”

[the remark this answers is not in the document]
Comment (brancz) [reply replicate.go]: That’s a little more aggressive than I like my software to be haha

[the remark this answers is not in the document]
Comment (bwplotka) [reply replicate.go]: one-off or maybe wait (:

[the remark this answers is not in the document]
Comment (bwplotka) [reply replicate.go]: doing `one-off`

**Pull request merged**